File: Whatsapp/OpenWA/sdk/javascript/src/resources/health.ts
/**
 * Health resource — connectivity and readiness probes.
 *
 * Backed by `src/modules/health/health.controller.ts`.
 * @packageDocumentation
 */

import { encodeSegment } from '../http.js';
import type { OpenWAClient } from '../client.js';
import type { HealthReadyResponse, HealthResponse } from '../types.js';

export class HealthResource {
  constructor(private readonly client: OpenWAClient) {}

  /** General health (also returns the running version). */
  check(): Promise<HealthResponse> {
    return this.client.request<HealthResponse>({ method: 'GET', path: '/api/health' });
  }

  /** Kubernetes liveness probe. */
  live(): Promise<{ status: string }> {
    return this.client.request<{ status: string }>({ method: 'GET', path: '/api/health/live' });
  }

  /** Kubernetes readiness probe — checks both DB connections. */
  ready(): Promise<HealthReadyResponse> {
    return this.client.request<HealthReadyResponse>({ method: 'GET', path: '/api/health/ready' });
  }
}

File: OpenWA/sdk/python/openwa/resources/health.py
"""Health resource — connectivity and readiness probes.

Backed by ``src/modules/health/health.controller.ts``.
"""

from __future__ import annotations

from typing import TYPE_CHECKING

from ..types import HealthReadyResponse, HealthResponse

if TYPE_CHECKING:
    from .._http import HttpExecutor


class HealthResource:
    def __init__(self, http: "HttpExecutor") -> None:
        self._http = http

    def check(self) -> HealthResponse:
        return self._http.request("GET", "/api/health")

    def live(self) -> dict[str, str]:
        return self._http.request("GET", "/api/health/live")

    def ready(self) -> HealthReadyResponse:
        return self._http.request("GET", "/api/health/ready")
